spec/features/directors_feature_spec.rb

```diff
@@ -0,0 +1,165 @@
+
+      directors = Director.all
+      directors.each do |director|
+        expect(page).to have_content(director.name)
+        expect(page).to have_content(director.dob)
+      end
+    end
+
+    it "displays a functional delete link for each director", points: 5 do
+      visit "/directors"
+
+      count_of_directors = Director.count
+
+      click_link 'Delete', match: :first
+
+      expect(Director.count).to eq(count_of_directors - 1)
+    end
+  end
+
+  context "show page" do
+    it "displays a count of the director's movies", points: 5 do
+      directors = Director.all
+      directors.each do |director|
+        visit "/directors/#{director.id}"
+
+        expect(page).to have_content(director.movies.count)
+      end
+    end
+
+    it "displays a list of the director's movies", points: 5 do
+      directors = Director.all
+      directors.each do |director|
+        visit "/directors/#{director.id}"
+
+        director.movies.each do |movie|
+          expect(page).to have_content(movie.title)
+        end
+      end
+    end
+
+    it "displays a form to add a new movie", points: 10 do
+      directors = Director.all
+      directors.each do |director|
+        visit "/directors/#{director.id}"
+
+        expect(page).to have_selector("form")
+      end
+    end
+
+    it "reates a new movie after submitting the form", points: 10 do
+      director = Director.last
+
+      visit "/directors/#{director.id}"
+
+      expect(page).to have_selector("form")
+
+      count_of_movies = Movie.count
+
+      fill_in 'Title', with: 'Mad Max: Fury Road'
+      fill_in 'Year', with: 2015
+      click_button 'Create Movie'
+
+      expect(Movie.count).to eq(count_of_movies + 1)
+    end
+
+    it "displays a hidden input field to associate a new movie to the director", points: 10 do
+      directors = Director.all
+      directors.each do |director|
+        visit "/directors/#{director.id}"
+
+        expect(page).to have_selector("input[value='#{director.id}']", visible: false),
+          "expected to find a hidden input field with the director's id as the value"
+      end
+    end
+  end
+
+  context "new form" do
+    it 'creates a new director after submitting the form' do
+      visit "/directors/new"
+
+      expect(page).to have_selector("form")
+
+      count_of_directors = Director.count
+
+      fill_in 'Name', with: 'George Miller'
+      fill_in 'Dob', with: 'March 3, 1945'
+      click_button 'Create Director'
+
+      expect(Director.count).to eq(count_of_directors + 1)
+    end
+
+    it 'fails the validation if name is not present' do
+      visit "/directors/new"
+
+      expect(page).to have_selector("form")
+
+      count_of_directors = Director.count
+
+      fill_in 'Name', with: ''
+      fill_in 'Dob', with: 'March 3, 1945'
+      click_button 'Create Director'
+
+      expect(Director.count).to eq(count_of_directors)
+    end
+
+    it 'fails the validation if name is not unique' do
+      visit "/directors/new"
```

Comment:
I think in the past I've written unit tests and you encouraged us to only use integration tests, so was just following that methodology.  Unit tests would be more appropriate for these validation tests, but the benefit of solely using integration testing is that it will be easier for students to repeat the tests in their local apps to see why they're failing.